### PR TITLE
Implement search feature for filtering groups by name

### PR DESCRIPTION
**Pull request by juliasliu**

### CODE REVIEW

**Review by ajs11893 — COMMENTED** [review]
A little confused about what the search function should do, clicking search doesn't seem to affect the graphic.
Number of groups changes, but number of unique users doesn't
Also, the text seems really small and when I click on a circle with no elements the text dissapears which can be a bit confusing.

**Review by ajs11893 — COMMENTED** [review]
I think in the loadSidebar() you need to update num-total-members

**Review by ajs11893 — COMMENTED** [review]
Ok seems like search works as expected sometimes, if I search "class" it just shows one group, but if I search "berkeley" it shows all groups, but it says the number of groups is 3, which is what I would expect.

**Comment by juliasliu** [comment]
Okay I resolved the search error but I have tabled the hovering feature for MVP #2. Note: If you test the filter by user component, it won't work--I have been trying to debug why I can't use memberKey to query groups but couldn't figure it out.

**Comment by juliasliu** [comment]
Note: You cannot search for group name or email AND filter by memberKey at the same time.

File: src/main/webapp/pages/group.html [hunk 1]
```diff
@@ -62,19 +63,79 @@
...
+
+                        <button type="button" class="btn btn-light" onclick="clearSidebar()">
+                            Clear all
+                            <i class="fa fa-times"></i>
+                        </button>
+                    </div>
+                    <div class="sidenav-section blue">
                         <div class="section-title">Key</div>
-                        <p></p>
+                        <p>loren ipsum</p>
```
Comment (ajs11893) [inline src/main/webapp/pages/group.html:138]: I'm guessing this text isn't shown?

File: src/main/webapp/scripts/group-script.js [hunk 2]
```diff
@@ -42,9 +91,88 @@ function loadSidebar() {
...
+    document.getElementById("user-sel").value = searchMemberKey;
+    document.getElementById("order-by-sel").value = orderBy;
+    document.getElementById("view-total-groups-sel").value = viewTotal;
+
+    getAllGroups();
+}
+
+function selectUser() {
+    var userSel = document.getElementById("user-sel");
+    if (userSel.value == "null") searchMemberKey = null;
```
Comment (ajs11893) [inline src/main/webapp/scripts/group-script.js]: Probably best to fully expand these if and else statements using curly brackets

File: src/main/webapp/scripts/group-script.js [hunk 3]
```diff
@@ -166,29 +294,36 @@ function visualize() {
         };
     }
 
-    var chartElement = document.getElementById("chart")
-    chartElement.appendChild(svg.node())
+    var chartElement = document.getElementById("chart");
+    if (chartElement.lastChild) chartElement.removeChild(chartElement.lastChild);
```
Comment (ajs11893) [inline src/main/webapp/scripts/group-script.js]: Expand this if statement too

File: src/main/webapp/scripts/group-script.js [hunk 4]
```diff
@@ -202,13 +337,25 @@ async function loadGroups() {
...
+                        group = groups[indexOfGroup];
+                    }
                     newCircle.children.push({
                         name: group.name,
                         value: parseInt(group.directMembersCount),
                         id: group.id
                     })
                 } else if (type == "USER") {
+                    var indexOfUser = users.findIndex(elem => elem.id == members[j].id)
+                    if (indexOfUser < 0) {
```
Comment (ajs11893) [inline src/main/webapp/scripts/group-script.js:362]: add comment noting that indexOfUser will be less than 0 (i believe -1) if it is not present

**Review by ajs11893 — APPROVED** [review]
Looks good

**Pull request merged**